File: src/AAS.Domain/Bids/BidBlank.cs
#region

using AAS.Domain.Bids.Enums;
using AAS.Domain.Files;
using AAS.Tools.Types.IDs;

#endregion

namespace AAS.Domain.Bids;

public class BidBlank
{
    public ID? Id { get; set; }
    public int? Number { get; set; }
    public string? Title { get; set; }
    public string? Description { get; set; }
    public string? DenyDescription { get; set; }
    public FileBlank[] FileBlanks { get; set; }
    public BidStatus Status { get; set; }
    public DateOnly? ApproximateDate { get; set; }
    public ID CreatedUserId { get; set; }

    public BidBlank(ID? id, int? number, string? title, string? description, string? denyDescription, FileBlank[] fileBlanks, BidStatus status,
        DateOnly? approximateDate, ID createdUserId)
    {
        Id = id;
        Number = number;
        Title = title;
        Description = description;
        DenyDescription = denyDescription;
        FileBlanks = fileBlanks;
        Status = status;
        ApproximateDate = approximateDate;
        CreatedUserId = createdUserId;
    }
}

File: src/AAS.BackOffice/client_app/src/domain/bids/bidBlank.ts
import { FileBlank } from "../files/fileBlank"
import { Bid } from "./bid"
import { BidStatus } from "./bidStatus"

export interface BidBlank {
    id: string | null,
    number: number | null,
    title: string | null,
    description: string | null,
    denyDescription: string | null,
    status: BidStatus,
    acceptanceDate: Date | null,
    approximateDate: string | Date | null | undefined,
    fileBlanks: FileBlank[],
    createdUserId: string
}

export namespace BidBlank {
    export function getDefault(systemUserId: string): BidBlank {
        return {
            id: null,
            number: null,
            title: null,
            description: null,
            denyDescription: null,
            status: BidStatus.AwaitingVerification,
            acceptanceDate: null,
            approximateDate: null,
            fileBlanks: [],
            createdUserId: systemUserId
        }
    }

    export function fromBid(bid: Bid): BidBlank {
        return {
            id: bid.id,
            number: bid.number,
            title: bid.title,
            description: bid.description,
            denyDescription: bid.denyDescription,
            status: bid.status,
            acceptanceDate: bid.acceptanceDate,
            approximateDate: bid.approximateDate,
            fileBlanks: bid.files.map(file => FileBlank.fromFile(file)),
            createdUserId: bid.createdUserId
        }
    }
}